public/js/create-a-set.js

```diff
@@ -0,0 +1,157 @@
+"use strict";
+// Module pattern for encapsulation
+(function() {
+  const firebaseConfig = {
+    apiKey: "AIzaSyAH7XCKlraO6k5zeoW9jCpqPgQwXBSydhw",
+    authDomain: "tester-7bc61.firebaseapp.com",
+    databaseURL: "https://tester-7bc61.firebaseio.com",
+    projectId: "tester-7bc61",
+    storageBucket: "tester-7bc61.appspot.com",
+    messagingSenderId: "244317548908",
+    appId: "1:244317548908:web:3bde08538be9dda7"
+  };
+
+  function $(id) {
+    return document.getElementById(id);
+  }
+
+  /**
+   * The ID of the currently signed-in User. We keep track of this to detect Auth state change events that are just
+   * programmatic token refresh but not a User status change.
+   */
+  var currentUID;
+
+  /**
+   * Triggers every time there is a change in the Firebase auth state (i.e. user signed-in or user signed out).
+   */
+  function onAuthStateChanged(user) {
+    // We ignore token refresh events.
+    if (user && currentUID === user.uid) {
+      return;
+    }
+    if (user) {
+      // User is signed in.
+      currentUID = user.uid;
+      var displayName = user.displayName;
+      var email = user.email;
+      var emailVerified = user.emailVerified;
+      var photoURL = user.photoURL;
+      var isAnonymous = user.isAnonymous;
+      var uid = user.uid;
+      var providerData = user.providerData;
+      $("quickstart-sign-in-status").textContent = "Signed in";
+    } else {
+      // User is signed out.
+      attemptRedirect();
+    }
+  }
+
+  // Replace confirm with alert then directly replace window location.
+  // Will also need to change window location to whatever url of the sign in page is.
+  function attemptRedirect() {
+    let confirmAnswer = confirm(
+      "You're signed out, you'll be redirected to the sign in page."
+    );
+    if (confirmAnswer) {
+      window.location.href = "/index.html";
+    } else {
+      alert(
+        "We probably should just redirect them, but that's annoying for testing."
+      );
+    }
+  }
+
+  // Not sure we need to or can validate UTF-8 string but checks length on client side.
```

Comment:
If the page is being encoded as UTF-8 with `<meta charset=utf-8 />`, then all input should also be automatically encoded as UTF-8.